i18n-helpers/src/lib.rs

```diff
@@ -603,25 +702,37 @@ pub fn reconstruct_markdown(
 ///     ],
 /// );
 /// ```
-pub fn extract_messages(document: &str) -> Vec<(usize, String)> {
+pub fn extract_messages(document: &str) -> Vec<(usize, ExtractedMessage)> {
     let events = extract_events(document, None);
     let mut messages = Vec::new();
     let mut state = None;
 
     for group in group_events(&events) {
         match group {
-            Group::Translate(events) => {
+            Group::Translate(events, group_extra) => {
                 if let Some((lineno, _)) = events.first() {
                     let (text, new_state) = reconstruct_markdown(&events, state);
                     // Skip empty messages since they are special:
                     // they contains the PO file metadata.
                     if !text.trim().is_empty() {
-                        messages.push((*lineno, text));
+                        messages.push((
+                            *lineno,
+                            ExtractedMessage {
+                                message: text,
+                                translator_comment: {
+                                    if group_extra.translator_comments.is_empty() {
+                                        None
+                                    } else {
+                                        Some(group_extra.translator_comments.join(" "))
+                                    }
```

Comment:
Consider using the little-known [bool::then](https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then) method here:

```suggestion
                                    group_extra.translator_comments
                                        .is_empty()
                                        .then(|| group_extra.translator_comments.join(" "))
```

Alternatively, change `translator_comment` to `String` and model the absence of one with an empty string. As long as polib skips such a comment, this would make your life easier.

Reply:
Okay, I will simplify to using just a `String` rather than an `Option<String>`; I think I was probably concerned about presence detection for comments when I first wrote this.  But empty string is handled as expected, so I don't need the presence detection after all.